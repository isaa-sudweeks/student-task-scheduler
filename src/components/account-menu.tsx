"use client";

import React from "react";
import { useSession, signOut } from "next-auth/react";
import Link from "next/link";
import { usePathname, useSearchParams } from "next/navigation";

export function AccountMenu() {
  const { data } = useSession();
  const user = data?.user;
  const pathname = usePathname();
  const searchParams = useSearchParams();
  const current = React.useMemo(() => {
    const q = searchParams?.toString();
    return q && q.length > 0 ? `${pathname}?${q}` : pathname;
  }, [pathname, searchParams]);
  const initials = React.useMemo(() => {
    if (!user?.name) return "";
    const parts = user.name.split(" ").filter(Boolean);
    return parts.slice(0, 2).map(p => p[0]?.toUpperCase()).join("");
  }, [user?.name]);

  const [open, setOpen] = React.useState(false);
  const menuRef = React.useRef<HTMLDivElement | null>(null);

  React.useEffect(() => {
    const onDocClick = (e: MouseEvent) => {
      if (!menuRef.current) return;
      if (!menuRef.current.contains(e.target as Node)) {
        setOpen(false);
      }
    };
    document.addEventListener("mousedown", onDocClick);
    return () => document.removeEventListener("mousedown", onDocClick);
  }, []);

  return (
    <div className="relative" ref={menuRef}>
      <button
        type="button"
        aria-haspopup="menu"
        aria-expanded={open}
        onClick={() => setOpen((v) => !v)}
        className="inline-flex h-9 w-9 items-center justify-center rounded-full border border-transparent p-0 hover:bg-black/5 dark:hover:bg-white/10"
        aria-label="Account menu"
        title={user?.name ? `${user.name} â€” Account menu` : "Account menu"}
      >
        {user?.image ? (
          // eslint-disable-next-line @next/next/no-img-element
          <img src={user.image} alt={user?.name ?? "Account"} className="h-9 w-9 rounded-full object-cover" />
        ) : (
          <span className="inline-flex h-9 w-9 items-center justify-center rounded-full bg-black/10 text-xs dark:bg-white/10">
            {initials || ""}
          </span>
        )}
      </button>

      {open && (
        <div
          role="menu"
          aria-label="Account menu"
          className="absolute right-0 z-50 mt-2 w-56 rounded border bg-white p-1 text-sm shadow-lg dark:border-white/10 dark:bg-zinc-900"
        >
          <Link
            role="menuitem"
            href={`/settings?returnTo=${encodeURIComponent(current || "/")}`}
            onClick={() => setOpen(false)}
            className="block w-full rounded px-3 py-2 text-left hover:bg-black/5 dark:hover:bg-white/10"
            title="Account settings"
          >
            Account Settings
          </Link>
          <button
            role="menuitem"
            type="button"
            onClick={() => {
              setOpen(false);
              signOut({ callbackUrl: "/api/auth/signin" });
            }}
            className="block w-full rounded px-3 py-2 text-left hover:bg-black/5 dark:hover:bg-white/10"
            title="Sign out"
          >
            Sign out
          </button>
        </div>
      )}
    </div>
  );
}
