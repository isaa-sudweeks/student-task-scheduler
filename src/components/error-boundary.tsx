"use client";
import React from "react";

type ErrorBoundaryProps = {
  children: React.ReactNode;
  fallback?: React.ReactNode;
};

type ErrorBoundaryState = {
  hasError: boolean;
};

export class ErrorBoundary extends React.Component<
  ErrorBoundaryProps,
  ErrorBoundaryState
> {
  state: ErrorBoundaryState = { hasError: false };

  static getDerivedStateFromError(): ErrorBoundaryState {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    console.error(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return this.props.fallback ?? (
        <div role="alert">Something went wrong</div>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
