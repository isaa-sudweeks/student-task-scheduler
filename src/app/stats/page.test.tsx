// @vitest-environment jsdom
import React from 'react';
import { render, screen, cleanup, within } from '@testing-library/react';
import { describe, it, expect, vi, afterEach, beforeAll, beforeEach } from 'vitest';
import * as matchers from '@testing-library/jest-dom/matchers';

vi.mock('@/server/api/react', () => ({
  api: {
    task: {
      list: {
        useQuery: vi.fn(),
      },
    },
    focus: {
      aggregate: {
        useQuery: vi.fn(),
      },
    },
  },
}));

let mockTheme = 'light';
vi.mock('next-themes', () => ({
  useTheme: () => ({ resolvedTheme: mockTheme }),
}));

import { api } from '@/server/api/react';
import StatsPage from './page';
import { ErrorBoundary } from '@/components/error-boundary';

const taskUseQueryMock = api.task.list.useQuery as ReturnType<typeof vi.fn>;
const focusUseQueryMock = api.focus.aggregate.useQuery as ReturnType<typeof vi.fn>;

expect.extend(matchers);

beforeAll(() => {
  class ResizeObserver {
    callback: ResizeObserverCallback;
    constructor(callback: ResizeObserverCallback) {
      this.callback = callback;
    }
    observe() {
      this.callback(
        [
          {
            contentRect: { width: 800, height: 400 },
          } as unknown as ResizeObserverEntry,
        ],
        this
      );
    }
    unobserve() {}
    disconnect() {}
  }
  (global as any).ResizeObserver = ResizeObserver;
});

afterEach(() => {
  cleanup();
  vi.clearAllMocks();
  mockTheme = 'light';
});

describe('StatsPage', () => {
  it('renders summary metrics', () => {
    taskUseQueryMock.mockReturnValue({
      data: [
        { id: '1', status: 'TODO', subject: 'Math', title: 'Task 1' },
        { id: '2', status: 'DONE', subject: 'Science', title: 'Task 2' },
        { id: '3', status: 'DONE', subject: 'Math', title: 'Task 3' },
      ],
      isLoading: false,
    });
    focusUseQueryMock.mockReturnValue({
      data: [{ taskId: '2', durationMs: 120000 }],
      isLoading: false,
    });

    render(<StatsPage />);
    const range = taskUseQueryMock.mock.calls[0][0];
    expect(range.start).toBeInstanceOf(Date);
    expect(range.end).toBeInstanceOf(Date);
    expect(focusUseQueryMock.mock.calls[0][0]).toBe(range);

    const totalCardLabel = screen.getByText('Total Tasks');
    const totalCard = totalCardLabel.parentElement?.parentElement as HTMLElement;
    expect(totalCard).toBeInTheDocument();
    expect(within(totalCard).getByText('3')).toBeInTheDocument();
    expect(totalCard.querySelector('svg')).toBeTruthy();

    const rateCardLabel = screen.getByText('Completion Rate');
    const rateCard = rateCardLabel.parentElement?.parentElement as HTMLElement;
    expect(rateCard).toBeInTheDocument();
    expect(within(rateCard).getByText('67%')).toBeInTheDocument();
    expect(rateCard.querySelector('svg')).toBeTruthy();

    const avgFocusLabel = screen.getByText('Avg Focus (m)');
    const avgFocusCard = avgFocusLabel.parentElement?.parentElement as HTMLElement;
    expect(avgFocusCard).toBeInTheDocument();
    expect(within(avgFocusCard).getByText('2')).toBeInTheDocument();
    expect(avgFocusCard.querySelector('svg')).toBeTruthy();
    expect(screen.getByText('TODO: 1')).toBeInTheDocument();
    expect(screen.getByText('DONE: 2')).toBeInTheDocument();
    expect(screen.getByText('Math: 2')).toBeInTheDocument();
    expect(screen.getByText('Science: 1')).toBeInTheDocument();
    expect(screen.getByText('Task 2: 2m')).toBeInTheDocument();
    expect(screen.getByText('Science: 2m')).toBeInTheDocument();
  });

  it('renders focus stats for tasks with duplicate titles without key warnings', () => {
    const consoleErrorSpy = vi
      .spyOn(console, 'error')
      .mockImplementation(() => {});
    taskUseQueryMock.mockReturnValue({
      data: [
        { id: '1', status: 'TODO', subject: 'Math', title: 'Task' },
        { id: '2', status: 'DONE', subject: 'Science', title: 'Task' },
      ],
      isLoading: false,
    });
    focusUseQueryMock.mockReturnValue({
      data: [
        { taskId: '1', durationMs: 60000 },
        { taskId: '2', durationMs: 120000 },
      ],
      isLoading: false,
    });

    render(<StatsPage />);
    const range = taskUseQueryMock.mock.calls[0][0];
    expect(range.start).toBeInstanceOf(Date);
    expect(range.end).toBeInstanceOf(Date);
    expect(focusUseQueryMock.mock.calls[0][0]).toBe(range);
    expect(screen.getByText('Task: 1m')).toBeInTheDocument();
    expect(screen.getByText('Task: 2m')).toBeInTheDocument();
    expect(screen.getByText('Math: 1m')).toBeInTheDocument();
    expect(screen.getByText('Science: 2m')).toBeInTheDocument();
    expect(consoleErrorSpy).not.toHaveBeenCalled();
    consoleErrorSpy.mockRestore();
  });

  it('renders fallback when query fails', () => {
    taskUseQueryMock.mockReturnValue({
      data: [],
      isLoading: false,
      error: new Error('oops'),
    });
    focusUseQueryMock.mockReturnValue({ data: [], isLoading: false });

    render(
      <ErrorBoundary fallback={<main>Failed to load stats</main>}>
        <StatsPage />
      </ErrorBoundary>
    );
    const range = taskUseQueryMock.mock.calls[0][0];
    expect(range.start).toBeInstanceOf(Date);
    expect(range.end).toBeInstanceOf(Date);
    expect(focusUseQueryMock.mock.calls[0][0]).toBe(range);
    expect(screen.getByText('Failed to load stats')).toBeInTheDocument();
  });

  describe('visual regression', () => {
    const tasks = [
      { id: '1', status: 'TODO', subject: 'Math', title: 'Task 1' },
      { id: '2', status: 'DONE', subject: 'Science', title: 'Task 2' },
    ];

    beforeEach(() => {
      taskUseQueryMock.mockReturnValue({
        data: tasks,
        isLoading: false,
      });
      focusUseQueryMock.mockReturnValue({
        data: [
          { taskId: '1', durationMs: 60000 },
          { taskId: '2', durationMs: 120000 },
        ],
        isLoading: false,
      });
    });

    it('matches light theme snapshot', () => {
      mockTheme = 'light';
      const { container } = render(<StatsPage />);
      expect(container).toMatchSnapshot();
    });

    it('matches dark theme snapshot', () => {
      mockTheme = 'dark';
      const { container } = render(<StatsPage />);
      expect(container).toMatchSnapshot();
    });
  });
});
