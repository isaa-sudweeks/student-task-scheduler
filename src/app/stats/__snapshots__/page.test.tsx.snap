// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatsPage > visual regression > matches dark theme snapshot 1`] = `
<div>
  <main
    class="space-y-6 text-neutral-900 dark:text-neutral-100"
  >
    <header
      class="flex flex-col gap-4 sm:flex-row sm:items-center sm:justify-between"
    >
      <h1
        class="text-2xl font-semibold"
      >
        Task Statistics
      </h1>
      <div
        class="flex items-center gap-3"
      >
        <div
          aria-label="Task filter"
          class="flex w-full flex-wrap items-center justify-center gap-2 md:justify-start"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-black dark:border-white"
            role="tab"
            type="button"
          >
            All
          </button>
          <button
            aria-selected="false"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-transparent"
            role="tab"
            type="button"
          >
            Overdue
          </button>
          <button
            aria-selected="false"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-transparent"
            role="tab"
            type="button"
          >
            Today
          </button>
          <button
            aria-selected="false"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-transparent"
            role="tab"
            type="button"
          >
            Archive
          </button>
          <div
            class="relative w-full sm:ml-2 sm:w-auto"
          >
            <svg
              class="lucide lucide-tag pointer-events-none absolute left-2 top-1/2 h-4 w-4 -translate-y-1/2 text-slate-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
              />
              <circle
                cx="7.5"
                cy="7.5"
                fill="currentColor"
                r=".5"
              />
            </svg>
            <select
              aria-label="Subject filter"
              class="w-full appearance-none rounded-full border border-slate-200 bg-slate-100 py-1.5 pl-8 pr-8 text-sm text-slate-800 shadow-sm transition hover:bg-slate-50 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500 focus-visible:ring-offset-2 dark:border-slate-700 dark:bg-slate-800/60 dark:text-slate-200 dark:hover:bg-slate-800 sm:w-auto"
              title="Filter by subject"
            >
              <option
                value=""
              >
                All subjects
              </option>
              <option
                value="Math"
              >
                Math
              </option>
              <option
                value="Science"
              >
                Science
              </option>
            </select>
            <svg
              class="lucide lucide-chevron-down pointer-events-none absolute right-2 top-1/2 h-4 w-4 -translate-y-1/2 text-slate-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </div>
        </div>
        <button
          class="inline-flex items-center justify-center rounded px-4 py-2 text-sm font-medium focus:outline-none disabled:opacity-50 disabled:cursor-not-allowed transition-colors bg-black text-white dark:bg-white dark:text-black"
        >
          Export CSV
        </button>
      </div>
    </header>
    <section
      class="flex items-end gap-2"
    >
      <label
        class="flex flex-col text-sm"
      >
        <span>
          Start
        </span>
        <input
          class="h-9 w-full rounded-md border border-black/10 bg-transparent px-3 text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500 focus-visible:ring-offset-2 dark:border-white/10"
          type="date"
          value="2025-10-11"
        />
      </label>
      <label
        class="flex flex-col text-sm"
      >
        <span>
          End
        </span>
        <input
          class="h-9 w-full rounded-md border border-black/10 bg-transparent px-3 text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500 focus-visible:ring-offset-2 dark:border-white/10"
          type="date"
          value="2025-10-18"
        />
      </label>
    </section>
    <section
      class="grid grid-cols-1 gap-4 sm:grid-cols-3"
    >
      <div
        class="flex items-center gap-3 rounded-md bg-white p-4 shadow dark:bg-neutral-800"
      >
        <svg
          class="lucide lucide-list h-6 w-6 text-indigo-600 dark:text-indigo-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="8"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="8"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="8"
            x2="21"
            y1="18"
            y2="18"
          />
          <line
            x1="3"
            x2="3.01"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="3.01"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="3.01"
            y1="18"
            y2="18"
          />
        </svg>
        <div
          class="flex flex-col"
        >
          <span
            class="text-sm text-neutral-600 dark:text-neutral-400"
          >
            Total Tasks
          </span>
          <span
            class="text-xl font-semibold text-neutral-900 dark:text-neutral-100"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="flex items-center gap-3 rounded-md bg-white p-4 shadow dark:bg-neutral-800"
      >
        <svg
          class="lucide lucide-circle-check-big h-6 w-6 text-green-600 dark:text-green-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.801 10A10 10 0 1 1 17 3.335"
          />
          <path
            d="m9 11 3 3L22 4"
          />
        </svg>
        <div
          class="flex flex-col"
        >
          <span
            class="text-sm text-neutral-600 dark:text-neutral-400"
          >
            Completion Rate
          </span>
          <span
            class="text-xl font-semibold text-neutral-900 dark:text-neutral-100"
          >
            50%
          </span>
        </div>
      </div>
      <div
        class="flex items-center gap-3 rounded-md bg-white p-4 shadow dark:bg-neutral-800"
      >
        <svg
          class="lucide lucide-list h-6 w-6 text-indigo-600 dark:text-indigo-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="8"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="8"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="8"
            x2="21"
            y1="18"
            y2="18"
          />
          <line
            x1="3"
            x2="3.01"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="3.01"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="3.01"
            y1="18"
            y2="18"
          />
        </svg>
        <div
          class="flex flex-col"
        >
          <span
            class="text-sm text-neutral-600 dark:text-neutral-400"
          >
            Avg Focus (m)
          </span>
          <span
            class="text-xl font-semibold text-neutral-900 dark:text-neutral-100"
          >
            2
          </span>
        </div>
      </div>
    </section>
    <div
      class="grid gap-6 md:grid-cols-2"
    >
      <section>
        <div
          class="space-y-2 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <h2
            class="text-xl font-medium"
          >
            By Status
          </h2>
          <ul>
            <li>
              TODO
              : 
              1
            </li>
            <li>
              DONE
              : 
              1
            </li>
          </ul>
          <div
            height="200"
            width="100%"
          >
            <div
              data="[object Object],[object Object]"
            />
          </div>
        </div>
      </section>
      <section>
        <div
          class="space-y-2 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <h2
            class="text-xl font-medium"
          >
            Focus Time by Subject
          </h2>
          <ul>
            <li>
              Math
              : 
              1
              m
            </li>
            <li>
              Science
              : 
              2
              m
            </li>
          </ul>
          <div
            height="200"
            width="100%"
          >
            <div
              data="[object Object],[object Object]"
            />
          </div>
        </div>
      </section>
      <section>
        <div
          class="space-y-2 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <h2
            class="text-xl font-medium"
          >
            By Subject
          </h2>
          <ul>
            <li>
              Math
              : 
              1
            </li>
            <li>
              Science
              : 
              1
            </li>
          </ul>
          <div
            height="200"
            width="100%"
          >
            <div />
          </div>
        </div>
      </section>
      <section>
        <div
          class="space-y-3 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <div
            class="space-y-1"
          >
            <h2
              class="text-xl font-medium"
            >
              Planned vs Actual by Task
            </h2>
            <p
              class="text-sm text-neutral-600 dark:text-neutral-400"
            >
              Compare estimated effort to recorded focus time to spot over- or under-spent work.
            </p>
          </div>
          <div
            class="grid gap-2 text-sm sm:grid-cols-3"
          >
            <div
              class="rounded-md bg-neutral-100 p-3 dark:bg-neutral-800"
            >
              <p
                class="text-neutral-500 dark:text-neutral-400"
              >
                Total planned
              </p>
              <p
                class="text-lg font-semibold text-neutral-900 dark:text-neutral-100"
              >
                90
                m
              </p>
            </div>
            <div
              class="rounded-md bg-neutral-100 p-3 dark:bg-neutral-800"
            >
              <p
                class="text-neutral-500 dark:text-neutral-400"
              >
                Total actual
              </p>
              <p
                class="text-lg font-semibold text-neutral-900 dark:text-neutral-100"
              >
                3
                m
              </p>
            </div>
            <div
              class="rounded-md bg-neutral-100 p-3 dark:bg-neutral-800"
            >
              <p
                class="text-neutral-500 dark:text-neutral-400"
              >
                Net delta
              </p>
              <p
                class="text-lg font-semibold text-neutral-900 dark:text-neutral-100"
              >
                -87
                m
              </p>
              <p
                class="mt-1 text-xs text-neutral-500 dark:text-neutral-400"
              >
                Over plan: 
                0
                 • Under plan: 
                2
              </p>
            </div>
          </div>
          <ul
            class="space-y-1 text-sm"
          >
            <li>
              Task 1
              : planned 
              30
              m • actual 
              1
              m • 
               
              -29
              m
            </li>
            <li>
              Task 2
              : planned 
              60
              m • actual 
              2
              m • 
               
              -58
              m
            </li>
          </ul>
          <div
            height="240"
            width="100%"
          >
            <div
              data="[object Object],[object Object]"
            />
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;

exports[`StatsPage > visual regression > matches light theme snapshot 1`] = `
<div>
  <main
    class="space-y-6 text-neutral-900 dark:text-neutral-100"
  >
    <header
      class="flex flex-col gap-4 sm:flex-row sm:items-center sm:justify-between"
    >
      <h1
        class="text-2xl font-semibold"
      >
        Task Statistics
      </h1>
      <div
        class="flex items-center gap-3"
      >
        <div
          aria-label="Task filter"
          class="flex w-full flex-wrap items-center justify-center gap-2 md:justify-start"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-black dark:border-white"
            role="tab"
            type="button"
          >
            All
          </button>
          <button
            aria-selected="false"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-transparent"
            role="tab"
            type="button"
          >
            Overdue
          </button>
          <button
            aria-selected="false"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-transparent"
            role="tab"
            type="button"
          >
            Today
          </button>
          <button
            aria-selected="false"
            class="bg-transparent px-3 py-1 text-sm border-b-2 transition border-transparent"
            role="tab"
            type="button"
          >
            Archive
          </button>
          <div
            class="relative w-full sm:ml-2 sm:w-auto"
          >
            <svg
              class="lucide lucide-tag pointer-events-none absolute left-2 top-1/2 h-4 w-4 -translate-y-1/2 text-slate-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
              />
              <circle
                cx="7.5"
                cy="7.5"
                fill="currentColor"
                r=".5"
              />
            </svg>
            <select
              aria-label="Subject filter"
              class="w-full appearance-none rounded-full border border-slate-200 bg-slate-100 py-1.5 pl-8 pr-8 text-sm text-slate-800 shadow-sm transition hover:bg-slate-50 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500 focus-visible:ring-offset-2 dark:border-slate-700 dark:bg-slate-800/60 dark:text-slate-200 dark:hover:bg-slate-800 sm:w-auto"
              title="Filter by subject"
            >
              <option
                value=""
              >
                All subjects
              </option>
              <option
                value="Math"
              >
                Math
              </option>
              <option
                value="Science"
              >
                Science
              </option>
            </select>
            <svg
              class="lucide lucide-chevron-down pointer-events-none absolute right-2 top-1/2 h-4 w-4 -translate-y-1/2 text-slate-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </div>
        </div>
        <button
          class="inline-flex items-center justify-center rounded px-4 py-2 text-sm font-medium focus:outline-none disabled:opacity-50 disabled:cursor-not-allowed transition-colors bg-black text-white dark:bg-white dark:text-black"
        >
          Export CSV
        </button>
      </div>
    </header>
    <section
      class="flex items-end gap-2"
    >
      <label
        class="flex flex-col text-sm"
      >
        <span>
          Start
        </span>
        <input
          class="h-9 w-full rounded-md border border-black/10 bg-transparent px-3 text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500 focus-visible:ring-offset-2 dark:border-white/10"
          type="date"
          value="2025-10-11"
        />
      </label>
      <label
        class="flex flex-col text-sm"
      >
        <span>
          End
        </span>
        <input
          class="h-9 w-full rounded-md border border-black/10 bg-transparent px-3 text-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500 focus-visible:ring-offset-2 dark:border-white/10"
          type="date"
          value="2025-10-18"
        />
      </label>
    </section>
    <section
      class="grid grid-cols-1 gap-4 sm:grid-cols-3"
    >
      <div
        class="flex items-center gap-3 rounded-md bg-white p-4 shadow dark:bg-neutral-800"
      >
        <svg
          class="lucide lucide-list h-6 w-6 text-indigo-600 dark:text-indigo-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="8"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="8"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="8"
            x2="21"
            y1="18"
            y2="18"
          />
          <line
            x1="3"
            x2="3.01"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="3.01"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="3.01"
            y1="18"
            y2="18"
          />
        </svg>
        <div
          class="flex flex-col"
        >
          <span
            class="text-sm text-neutral-600 dark:text-neutral-400"
          >
            Total Tasks
          </span>
          <span
            class="text-xl font-semibold text-neutral-900 dark:text-neutral-100"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="flex items-center gap-3 rounded-md bg-white p-4 shadow dark:bg-neutral-800"
      >
        <svg
          class="lucide lucide-circle-check-big h-6 w-6 text-green-600 dark:text-green-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.801 10A10 10 0 1 1 17 3.335"
          />
          <path
            d="m9 11 3 3L22 4"
          />
        </svg>
        <div
          class="flex flex-col"
        >
          <span
            class="text-sm text-neutral-600 dark:text-neutral-400"
          >
            Completion Rate
          </span>
          <span
            class="text-xl font-semibold text-neutral-900 dark:text-neutral-100"
          >
            50%
          </span>
        </div>
      </div>
      <div
        class="flex items-center gap-3 rounded-md bg-white p-4 shadow dark:bg-neutral-800"
      >
        <svg
          class="lucide lucide-list h-6 w-6 text-indigo-600 dark:text-indigo-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="8"
            x2="21"
            y1="6"
            y2="6"
          />
          <line
            x1="8"
            x2="21"
            y1="12"
            y2="12"
          />
          <line
            x1="8"
            x2="21"
            y1="18"
            y2="18"
          />
          <line
            x1="3"
            x2="3.01"
            y1="6"
            y2="6"
          />
          <line
            x1="3"
            x2="3.01"
            y1="12"
            y2="12"
          />
          <line
            x1="3"
            x2="3.01"
            y1="18"
            y2="18"
          />
        </svg>
        <div
          class="flex flex-col"
        >
          <span
            class="text-sm text-neutral-600 dark:text-neutral-400"
          >
            Avg Focus (m)
          </span>
          <span
            class="text-xl font-semibold text-neutral-900 dark:text-neutral-100"
          >
            2
          </span>
        </div>
      </div>
    </section>
    <div
      class="grid gap-6 md:grid-cols-2"
    >
      <section>
        <div
          class="space-y-2 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <h2
            class="text-xl font-medium"
          >
            By Status
          </h2>
          <ul>
            <li>
              TODO
              : 
              1
            </li>
            <li>
              DONE
              : 
              1
            </li>
          </ul>
          <div
            height="200"
            width="100%"
          >
            <div
              data="[object Object],[object Object]"
            />
          </div>
        </div>
      </section>
      <section>
        <div
          class="space-y-2 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <h2
            class="text-xl font-medium"
          >
            Focus Time by Subject
          </h2>
          <ul>
            <li>
              Math
              : 
              1
              m
            </li>
            <li>
              Science
              : 
              2
              m
            </li>
          </ul>
          <div
            height="200"
            width="100%"
          >
            <div
              data="[object Object],[object Object]"
            />
          </div>
        </div>
      </section>
      <section>
        <div
          class="space-y-2 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <h2
            class="text-xl font-medium"
          >
            By Subject
          </h2>
          <ul>
            <li>
              Math
              : 
              1
            </li>
            <li>
              Science
              : 
              1
            </li>
          </ul>
          <div
            height="200"
            width="100%"
          >
            <div />
          </div>
        </div>
      </section>
      <section>
        <div
          class="space-y-3 rounded-lg border p-4 shadow-sm bg-white dark:bg-neutral-900"
        >
          <div
            class="space-y-1"
          >
            <h2
              class="text-xl font-medium"
            >
              Planned vs Actual by Task
            </h2>
            <p
              class="text-sm text-neutral-600 dark:text-neutral-400"
            >
              Compare estimated effort to recorded focus time to spot over- or under-spent work.
            </p>
          </div>
          <div
            class="grid gap-2 text-sm sm:grid-cols-3"
          >
            <div
              class="rounded-md bg-neutral-100 p-3 dark:bg-neutral-800"
            >
              <p
                class="text-neutral-500 dark:text-neutral-400"
              >
                Total planned
              </p>
              <p
                class="text-lg font-semibold text-neutral-900 dark:text-neutral-100"
              >
                90
                m
              </p>
            </div>
            <div
              class="rounded-md bg-neutral-100 p-3 dark:bg-neutral-800"
            >
              <p
                class="text-neutral-500 dark:text-neutral-400"
              >
                Total actual
              </p>
              <p
                class="text-lg font-semibold text-neutral-900 dark:text-neutral-100"
              >
                3
                m
              </p>
            </div>
            <div
              class="rounded-md bg-neutral-100 p-3 dark:bg-neutral-800"
            >
              <p
                class="text-neutral-500 dark:text-neutral-400"
              >
                Net delta
              </p>
              <p
                class="text-lg font-semibold text-neutral-900 dark:text-neutral-100"
              >
                -87
                m
              </p>
              <p
                class="mt-1 text-xs text-neutral-500 dark:text-neutral-400"
              >
                Over plan: 
                0
                 • Under plan: 
                2
              </p>
            </div>
          </div>
          <ul
            class="space-y-1 text-sm"
          >
            <li>
              Task 1
              : planned 
              30
              m • actual 
              1
              m • 
               
              -29
              m
            </li>
            <li>
              Task 2
              : planned 
              60
              m • actual 
              2
              m • 
               
              -58
              m
            </li>
          </ul>
          <div
            height="240"
            width="100%"
          >
            <div
              data="[object Object],[object Object]"
            />
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;
